File: Preg2/fibonacci3-lambda-recursivo.hs
import System.IO
import Data.Function (fix)

main = do
    putStrLn "¿Cuantos elementos?: "
    x <- getLine
    let num = read x :: Integer
    print ([fix (\f -> (\n -> if n<=1 then 0 else if n > 3 then f (n-1) + f (n-2) + f (n-3) else 1)) n| n <- [1..num]])

File: Preg2/fibonacci3-lambda-recursivo.py
fiboelem = lambda n: 0 if n == 0 else 1 if n == 1 else 1 if n == 2 else fiboelem(n-1)+fiboelem(n-2)+fiboelem(n-3)

n=input("¿Cuantos elementos?: ")
n=int(n)
for i in range(0,n):
    print(fiboelem(i))